pom.xml

```diff
@@ -224,13 +225,37 @@
           <attach>true</attach>
           <excludePackageNames>javassist.compiler:javassist.convert:javassist.scopedpool:javassist.bytecode.stackmap</excludePackageNames>
           <bottom><![CDATA[<i>Javassist, a Java-bytecode translator toolkit.<br>
-Copyright (C) 1999- Shigeru Chiba. All Rights Reserved.</i>]]></bottom>
+Copyright (C) ${project.inceptionYear}- Shigeru Chiba. All Rights Reserved.</i>]]></bottom>
           <show>public</show>
           <nohelp>true</nohelp>
           <doclint>none</doclint>
           <source>8</source>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <version>3.4.1</version>
+        <executions>
+          <execution>
+            <id>enforces</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <!-- official docs: https://maven.apache.org/enforcer/enforcer-rules/index.html -->
+                <requireMavenVersion>
+                  <version>3.2.5</version>
+                </requireMavenVersion>
+                <requireJavaVersion>
+                  <version>11</version>
```

Comment:
Is this consistent with the other parts?  The binary is built by Java 8.

Reply:
`requireJavaVersion` enforce the java version to run maven. 
I found that compile failed with jdk 8 (and test target is 11).
Java 21 satisfies `requireJavaVersion 11`.

The binary version(byte code version) is controlled by `javac target` option.

-------

The guarantee that the built binary can used by java 8 application, is important for foundational library(javassit).

the CI can check this guarantee with some more work:

- build/compile with java 21 or 11(e.g. `maven install`)
- run test(without compile) on multiple java versions(e.g. `mvn surfire:test`)
  - the test case can not run by java 8, should be conditional skipped by test framework.

---------

I do this(build by java high version and run test by java low version in CI/GitHub action) in some projects.

I can try to add this to javassist CI/GitHub action.

Reply:
> I found that compile failed with jdk 8 (and test target is 11).

This is correct.  Some features are only for Java 11 and above.